apps/api/src/modules/job/job.service.ts

```diff
@@ -20,34 +23,127 @@ export class JobService {
-        //TODO: will find out how to filter range later [24]
-        // startDate: searchJobDto.startDate || undefined,
-        // startTime: searchJobDto.startTime || undefined,
-        // endDate: searchJobDto.endDate || undefined,
-        // endTime: searchJobDto.endTime || undefined,
-
-        //TODO: will find out how to filter const into range later [24]
-        // age: searchJobDto.age || undefined,
-
-        //TODO: will find out how to filter range later [24]
-        // minWage: searchJobDto.minWage || undefined,
-        // maxWage: searchJobDto.maxWage || undefined,
-
-        //TODO: will find out how to filter enum later [24]
-        //status: searchJobDto.status || undefined,
-        //gender: searchJobDto.gender || undefined,
-
+        Shooting: {
+          every: {
+            startDate: {
+              gte: searchJobDto.startDate || defaultStartDate,
+            },
+            endDate: {
+              lte: searchJobDto.endDate || defaultEndDate,
+            },
+            startTime: {
+              gte: searchJobDto.startTime || defaultStartTime,
+            },
+            endTime: {
+              lte: searchJobDto.endTime || defaultEndTime,
+            },
+          },
+          some: {
+            shootingLocation: searchJobDto.location || undefined,
+          },
+        },
+
+        minAge: {
+          lte: Number(searchJobDto.age) || defaultminAgeLte,
+        },
+        maxAge: {
+          gte: Number(searchJobDto.age) || defaultMaxAgeGte,
+        },
+
+        wage: {
+          gte: Number(searchJobDto.minWage) || defaultMinWageGte,
+          lte: Number(searchJobDto.maxWage) || defaultMaxWageLte,
+        },
+
+        status: undefined,
+        gender: undefined,
         castingId: Number(searchJobDto.castingId) || undefined,
       },
     }
+    //handle array and undefined
+    searchJobDto.status = searchJobDto.status || [SearchJobStatus.OPEN]
+    //check searchJobDto.status is array or not
+    if (!Array.isArray(searchJobDto.status)) {
+      searchJobDto.status = [searchJobDto.status]
+    }
+    let statusQuery = {}
+    if (
+      searchJobDto.status.includes(SearchJobStatus.OPEN) &&
+      searchJobDto.status.includes(SearchJobStatus.CLOSE)
+    ) {
+      statusQuery = [
+        JobStatus.OPEN,
+        JobStatus.SELECTING,
+        JobStatus.SELECTION_ENDED,
+        JobStatus.FINISHED,
+        JobStatus.CANCELLED,
+      ]
+    } else if (searchJobDto.status.includes(SearchJobStatus.OPEN)) {
+      statusQuery = [JobStatus.OPEN]
+    } else if (searchJobDto.status.includes(SearchJobStatus.CLOSE)) {
+      if (user.type == UserType.ACTOR) {
+        statusQuery = [
+          JobStatus.SELECTING,
+          JobStatus.SELECTION_ENDED,
+          JobStatus.FINISHED,
+        ]
+      } else {
+        statusQuery = [
+          JobStatus.SELECTING,
+          JobStatus.SELECTION_ENDED,
+          JobStatus.FINISHED,
+          JobStatus.CANCELLED,
+        ]
```

Comment:
Maybe this will reduce redundant 
```suggestion
        statusQuery = [
          JobStatus.SELECTING,
          JobStatus.SELECTION_ENDED,
          JobStatus.FINISHED,
        ]
      if (user.type == UserType.CASTING) statusQuery.push(JobStatus.CANCELLED)
```